spec/integration/parser/compiler_spec.rb

```diff
@@ -153,6 +153,17 @@ class c inherits b { notify { hi: } }
     end
   end
 
+  context 'when working with $settings name space' do
+    include PuppetSpec::Compiler
+    it 'makes $settings::strict available as string' do
+      node = Puppet::Node.new("testing")
+      catalog = compile_to_catalog(<<-MANIFEST, node)
+          notify { 'test': message => $settings::strict == 'warning' }
```

Comment:
does this need more spec tests/acceptance?
it looks to me (ignorantly) like this is only testing the String case in the switch statement in the PR.  i.e.: this spec change doesn't test the Symbol, Hash, Array cases (the fore-most being what the ticket is about)